r1cs/gadgets/std/src/fields/nonnative/reduce.rs

```diff
@@ -283,7 +339,7 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField> Reducer<SimulationF, Cons
     /// 
     // Costs
     // ``
-    //      (num_groups - 1) * (1 + bits_per_limb + surfeit + 2 - shift_per_limb) + 2
+    //      (num_groups - 1) * (1 + bits_per_limb + surfeit + 3 - shift_per_limb) + 2
```

Comment:
Was it wrong before ?

Reply:
Yes, if i recall correctly it was again due to the unpacking constraint that was not taken into account when converting each group to bits

Reply:
see below.